src/main/java/org/broadinstitute/hellbender/engine/filters/MarkedDuplicateReadFilter.java

```diff
@@ -0,0 +1,31 @@
+package org.broadinstitute.hellbender.engine.filters;
+
+import htsjdk.samtools.SAMFileHeader;
+import org.broadinstitute.hellbender.cmdline.Argument;
+import org.broadinstitute.hellbender.tools.spark.transforms.markduplicates.MarkDuplicatesSparkUtils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.read.ReadUtils;
+
+import java.io.Serializable;
+import java.util.HashSet;
+import java.util.Set;
+
+public final class MarkedDuplicateReadFilter extends ReadFilter implements Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    //Filters out optical duplicate reads (marked with the OD tag)
```

Comment:
The intended behavior is to remove optical duplicates. Firstly, PCR duplicates are detected using the position of the read alignment and we are not assuming the input BAM is aligned. Second, it is possible to have identical independent reads, particularly with RNAseq data, which is a possible use case.

I renamed the class to MarkedOpticalDuplicateReadFilter and corrected the behavior by looking for the OD tag and not looking at the flags.